doc/source/specification.rst

```diff
@@ -1,104 +1,112 @@
 Specification
 =============
-
-Description of how MorphIO sees morphologies.
-
-Formats:
---------
-MorphIO supports ASC (aka Neurolucida), H5 and SWC extensions
+Description of how MorphIO sees morphologies. MorphIO supports :ref:`.asc <specification-neurolucida>` (Neurolucida),
+`.h5`_ (H5 version 1) and `.swc`_ extensions.
 
 Soma formats:
 -------------
-(For more information, see: `Soma Format <http://neuromorpho.org/SomaFormat.html>`_\ ) MorphIO
-recognizes several kinds of soma format. The soma type is determined according to the morphology
-type and the number of data points:
+MorphIO implements Soma specification from `NeuroMorpho`_, and recognizes several kinds of soma format.
 
-Undefined soma SOMA_UNDEFINED
-*****************************
+No Soma
+*******
+MorphIO can read correctly a file without a soma. The soma type of such file will be ``SOMA_UNDEFINED``.
 
-MorphIO can read correctly a file without a soma. The latter will be of type "undefined".
-Additionnaly, a soma made of 2 points will be of type "undefined" as well. Indeed a 2-points soma
-does not make any sense in any of the other soma formats.
+Single point Soma
+*****************
 
-Soma single point SOMA_SINGLE_POINT
-***********************************
+.. image:: images/swc/sphere.svg
+   :scale: 100 %
+   :alt: soma single point
 
-If a soma is made of only one point it will be treated as a "single point soma" for H5 and ASC formats.
-The single point soma is represented as a sphere whose coordinate and radius are given by the soma point.
-SWC interprets such soma differently.
+If a soma is composed of only one point, it will be treated as a sphere with the center in this point, and the radius
```

Comment:
I don t know enough the format to be sure : in our morphologies + morphio do we make a difference between the single point contour and the single point ?

Reply:
[H5v1](https://developer.humanbrainproject.eu/docs/projects/morphology-documentation/0.0.2/h5v1.html) spec does not say anything about two point somas. According to it 2 point soma is not something illegal.

Reply:
That s true indeed but also not defined. So I guess having an undefined soma then makes sense ?

Reply:
> Somata represented as a single point in the original files are also converted into the three-point representation...

So this is what the neuromorph site does when it is ingesting morphologies - ie: they convert them to a 'standardised' representation.

I'm not sure whether ASC will output a single point, so we'll ignore that for now.

For H5, we should probably decide on what happens for 1 point, or 2 points, because they don't really define a contour - I think they should probably raise an exception.

Reply:
1-point and 2-point somas don't raise in H5 and ASC. An exception is only raised when `.surface` property is accessed. H5, ASC, SWC interpret 1-point as a sphere currently.

Reply:
I see, ok, that's really good info.

The soma fixes that @tomdele did for the `hdf5 reader cleanup` PR should fix the H5 version of the code.  Let's open a new issues for the ASC one, so we don't forget to tackle that.

Reply:
`hdf5 reader cleanup` PR didn't fix the H5 version. Two-point and one-point somas are still allowed for H5 and ASC. I think it is good to keep it as is. MorphIO is a plain reader. Because ASC and H5 do not explicitly say anything about such somas, we should not think for the specs, and define something on our own. Lets keep as is.